infrastructure/src/main/java/org/corfudb/infrastructure/management/FailureDetector.java

```diff
@@ -153,6 +149,9 @@ private PollReport pollRound(long epoch,
         // In each iteration we poll all the servers in the members list.
         for (int iteration = 0; iteration < failureThreshold; iteration++) {
 
+            final long pollIterationStartTime = System.nanoTime();
```

Comment:
I think my comment got lost from the last review. But why is this nanoTime?

Reply:
I think I replied to the initial comments. It's because of using relative time elapsed rather than time passed from epoch. all times are converted/used in millisecond though. Also more accurate.

Reply:
Ok, I remember now. But why is it necessary to have that precision (inter-millisecond), if you can't even use it? Converting back and forth just doesn't seem necessary.

Reply:
I agree with Maithem, we don't need nano time for network communication, it makes things more complicated

Reply:
@Maithem as mention on the first reply as well as above: 

> it's because of using relative time elapsed rather than time passed from epoch.

If you think that there is no difference between using nanoTime and currentTimeMillis in one being timer and the other wall clock, let me know.

(better accuracy of nanoTime even at the millisecond granularity is a side point and hence "also" in reply which was also mentioned as a side benefit but NOT the rationale for picking nanoTime the first reply).

Reply:
@xnull where in this code are we using nano time for network communication? Accuracy as mentioned several times is NOT the reason for using nanoTime. please read the comments above.

Reply:
I'm not asking you about the difference between System.currentTimeMillis and System.nanoTime, I think I understand that part.

Let me rephrase the question, how would using System.currentTimeMillis negatively affect the algorithm?

If the accuracy is the "side point", then what is the main point. Your argument is not clear. Furthermore, referring to previous points doesn't help either because you force pushed and they are no longer there for reference.

Reply:
@Maithem Ok, thanks for clarifying the concern. The main point is that using a timer I think is correct choice over using a wall clock. Example of negative effect: wrong benchmark when using currentTimeMillis after a clock sync which won't be the case with nanoTime.

The previous reply was addressing the same question (here it is, in case):
> nanoTime provides relative time rather than wall clock and is a better choice for micro benchmarking.

all in all, if you still prefer currentTimeMillis, I'll change nanoTime to it.

Reply:
The currentTimeMillis provides a wall clock time and is non monotonic which may lead to a regress as a result of clock adjustment.
nanoTime is a monotonic timer and won't regress.
We had an offline discussion to keep the nanoTime in order to avoid regression.

Reply:
> which may lead to a regress
BTW I didn't find anything about the regression in documentation http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/lang/System.java

Reply:
Related: https://bugs.java.com/bugdatabase/view_bug.do;jsessionid=c0af3bf36ff7fffffffffdbd3372166fe34d?bug_id=6311057

Reply:
@xnull 
from https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#currentTimeMillis--
>....Returns the **current time** in milliseconds. ....

hence a wall clock. As a result different calls to currentTimeMillis can potentially jump backward (or forward). 
on the other hand:
from https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#nanoTime--
> ...can only be used to measure elapsed time and is not related to any other notion of system or wall-clock time.

Reply:
@xnull what I quoted was copy pasted from java doc (links above).
Are you suggesting that using currentTimeMillis for timing can not lead to negative timers?